addons/payment_stripe_sca/models/payment.py

```diff
@@ -70,15 +72,42 @@ def _stripe_request(self, url, data=False, method="POST"):
 
     def _create_stripe_session(self, kwargs):
         self.ensure_one()
-        resp = self._stripe_request("checkout/sessions", kwargs)
-        if resp.get("payment_intent") and kwargs.get("client_reference_id"):
-            tx = (
-                self.env["payment.transaction"]
-                .sudo()
-                .search([("reference", "=", kwargs["client_reference_id"])])
-            )
-            tx.stripe_payment_intent = resp["payment_intent"]
-        return resp["id"]
+        resp = self._stripe_request('checkout/sessions', kwargs)
+        tx = False
+        if kwargs.get('client_reference_id'):
+            tx = self.env['payment.transaction'].sudo().search([('reference', '=', kwargs['client_reference_id'])])
+            if resp.get('payment_intent'):
+                tx.stripe_payment_intent = resp['payment_intent']
```

Comment:
How can I test this flow? I guess you are trying to fix a case where there is no `payment_intent` :thinking:

Reply:
Same remark: if you are trying to fix a bug => separate PR than this one that ought to log better error messages